java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -840,6 +855,28 @@ private void buildNewClass(NewClassTree tree) {
     build(Lists.reverse(tree.arguments()));
   }
 
+  private void handleExceptionalPaths(Symbol symbol) {
+    TryStatement tryStatement = enclosingTry.peek();
+    if(tryStatement != outerTry) {
+      currentBlock = createBlock(currentBlock);
+      currentBlock.exceptions.add(exitBlocks.peek());
+    }
+    if (symbol.isMethodSymbol()) {
+      List<Type> thrownTypes = ((Symbol.MethodSymbol) symbol).thrownTypes();
+      if (!thrownTypes.isEmpty()) {
+        thrownTypes.forEach(thrownType -> {
+          for (Type caughtType : tryStatement.catches.keySet()) {
+            if (thrownType.isSubtypeOf(caughtType)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)